src/graph/Graph.cpp

```diff
@@ -35,15 +35,26 @@ Vertex* Graph::findVertex(const Position &in) const {
     return NULL;
 }
 
+Vertex* Graph::findVertex(int idNode) const {
+    auto nodeTuple = this->vertexMap.find(idNode);
+    if(nodeTuple == this->vertexMap.end())
+        return NULL;
+    return nodeTuple->second;
+}
+
 /**
  *  Adds a vertex with a given content or info (in) to a graph (this).
  *  Returns true if successful, and false if a vertex with that content already exists.
  */
 
-bool Graph::addVertex(const Position &in) {
+bool Graph::addVertex(int id, const Position &in) {
     if ( findVertex(in) != NULL)
         return false;
-    vertexSet.push_back(new Vertex(in));
+    Vertex* newVertex = new Vertex(id, in);
+
+    vertexSet.push_back(newVertex); // Need to check if this vertexSet is needed
+    this->vertexMap.insert(pair<int, Vertex*>(id, newVertex));
+
```

Comment:
Isn't it easier to not worry about Ids here and just use the nextID functionality?

Reply:
Since the nodes file already includes information about IDs I decided to leave it here just for safety, since they could show up in non-crescent order

Reply:
Oh alright, that sounds great then

Reply:
Everything sounds great to me.